tests/pass-dep/concurrency/freebsd-futex.rs

```diff
@@ -0,0 +1,250 @@
+//@only-target: freebsd
+//@compile-flags: -Zmiri-preemption-rate=0 -Zmiri-disable-isolation
+
+use std::mem::{self, MaybeUninit};
+use std::ptr::{self, addr_of};
+use std::sync::atomic::AtomicU32;
+use std::time::{Duration, Instant};
+use std::{io, thread};
+
+fn wake_nobody() {
+    // Current thread waits on futex
+    // New thread wakes up 0 threads waiting on that futex
+    // Current thread should time out
+    static mut FUTEX: u32 = 0;
+
+    let waker = thread::spawn(|| {
+        thread::sleep(Duration::from_millis(200));
+
+        unsafe {
+            assert_eq!(
+                libc::_umtx_op(
+                    addr_of!(FUTEX) as *mut _,
+                    libc::UMTX_OP_WAKE_PRIVATE,
+                    0, // wake up 0 waiters
+                    ptr::null_mut::<libc::c_void>(),
+                    ptr::null_mut::<libc::c_void>(),
+                ),
+                0
+            );
+        }
+    });
+    let mut timeout = libc::timespec { tv_sec: 0, tv_nsec: 400_000_000 };
+    let timeout_size_arg =
+        ptr::without_provenance_mut::<libc::c_void>(mem::size_of::<libc::timespec>());
+    unsafe {
+        assert_eq!(
+            libc::_umtx_op(
+                addr_of!(FUTEX) as *mut _,
+                libc::UMTX_OP_WAIT_UINT_PRIVATE,
+                0,
```

Comment:
Is this the number of threads to wake?
What's the point of the sleep...? You disabled preemption, so this can be tested without `thread::sleep`.